client/src/main/java/io/pravega/client/stream/impl/ReaderGroupStateManager.java

```diff
@@ -332,10 +336,22 @@ private Duration calculateAcquireTime(ReaderGroupState state) {
     }
     
     String getCheckpoint() throws ReinitializationRequiredException {
+        fetchUpdatesIfNeeded();
         ReaderGroupState state = sync.getState();
+        long automaticCpInterval = state.getConfig().getAutomaticCheckpointIntervalMillis();
         if (!state.isReaderOnline(readerId)) {
             throw new ReinitializationRequiredException();
         }
+        String checkpoint = state.getCheckpointsForReader(readerId);
+        if (checkpoint != null) {
+            checkpointTimer.reset(Duration.ofMillis(automaticCpInterval));
+            return checkpoint;
+        }
+        if (automaticCpInterval <= 0 || checkpointTimer.hasRemaining() || state.hasOngoingCheckpoint()) {
+            return null;
+        }
+        sync.updateState(s -> s.hasOngoingCheckpoint() ? null : ImmutableList.of(new CreateCheckpoint()));
```

Comment:
1. That is an in-memory getter so it is cheap.
2. That could induce a race condition where if another checkpoint was added right before this one the two would be given out of order to the reader.